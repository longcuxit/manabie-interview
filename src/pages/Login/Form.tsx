import { usePushLoader } from "components/Loader";
import { Form, Button } from "react-bootstrap";
import { Link } from "react-router-dom";
import { useAuthAction } from "store/Auth";

export default function FormLogin() {
  const actions = useAuthAction();
  const pushLoader = usePushLoader();

  return (
    <Form
      onSubmit={(e) => {
        e.preventDefault();
        pushLoader(actions.login("username", "password"));
      }}
    >
      <Form.Group as="label" className="form-group mb-4 d-block">
        <Form.Control
          type="email"
          size="lg"
          placeholder="Enter a valid email address"
        />
        <Form.Label as="span">Email address</Form.Label>
      </Form.Group>

      <Form.Group as="label" className="form-group mb-4 d-block">
        <Form.Control type="password" size="lg" placeholder="Enter password" />
        <Form.Label as="span">Password</Form.Label>
      </Form.Group>

      <div className="d-flex justify-content-between align-items-center">
        <Form.Check label="Remember me" />

        <a href="#!" className="text-body">
          Forgot password?
        </a>
      </div>

      <div className="text-center text-lg-start mt-4 pt-2">
        <Button size="lg" className="px-2 w-100" type="submit">
          Login
        </Button>

        <p className="small fw-bold my-2 pt-1">
          Don't have an account?{" "}
          <Link to="#!" className="link-danger">
            Register
          </Link>
        </p>
      </div>
    </Form>
  );
}
