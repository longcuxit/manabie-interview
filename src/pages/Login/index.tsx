import { Container, Image } from "react-bootstrap";

import { LoaderWrapper } from "components/Loader";
import FormLogin from "./Form";

export default function Login() {
  return (
    <section className="vh-100 d-flex flex-column justify-content-between">
      <div />
      <Container fluid>
        <div className="row d-flex justify-content-center align-items-center h-100">
          <div className="col-md-9 col-lg-6 col-xl-5">
            <Image
              src="https://mdbcdn.b-cdn.net/img/Photos/new-templates/bootstrap-login-form/draw2.webp"
              fluid
              alt="Sample image"
            />
          </div>
          <div className="col-md-8 col-lg-6 col-xl-4 offset-xl-1">
            <FormLogin />
          </div>
        </div>
      </Container>
      <div className="d-flex flex-column flex-md-row text-center text-md-start justify-content-between py-4 px-4 px-xl-5 bg-primary">
        <div className="text-white mb-3 mb-md-0">
          Copyright Â© 2020. All rights reserved.
        </div>
      </div>
    </section>
  );
}
