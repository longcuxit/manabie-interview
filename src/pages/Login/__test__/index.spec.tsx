import { render } from "@testing-library/react";
import Login from "../index";

jest.mock("../Form", () => () => <>LoginForm</>);

test("pages/Login: match snapshot", () => {
  const { container } = render(<Login />);
  expect(container).toMatchSnapshot();
});
