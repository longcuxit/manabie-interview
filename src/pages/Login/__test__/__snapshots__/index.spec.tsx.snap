// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Login: match snapshot 1`] = `
<div>
  <section
    class="vh-100 d-flex flex-column justify-content-between"
  >
    <div />
    <div
      class="container-fluid"
    >
      <div
        class="row d-flex justify-content-center align-items-center h-100"
      >
        <div
          class="col-md-9 col-lg-6 col-xl-5"
        >
          <img
            alt="Sample image"
            class="img-fluid"
            src="https://mdbcdn.b-cdn.net/img/Photos/new-templates/bootstrap-login-form/draw2.webp"
          />
        </div>
        <div
          class="col-md-8 col-lg-6 col-xl-4 offset-xl-1"
        >
          LoginForm
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column flex-md-row text-center text-md-start justify-content-between py-4 px-4 px-xl-5 bg-primary"
    >
      <div
        class="text-white mb-3 mb-md-0"
      >
        Copyright Â© 2020. All rights reserved.
      </div>
    </div>
  </section>
</div>
`;
